src/main/scala/zio/rocksdb/transaction/package.scala

```diff
@@ -0,0 +1,41 @@
+package zio.rocksdb
+
+import org.{ rocksdb => jrocks }
+import zio._
+
+package object transaction {
+  type Bytes         = Array[Byte]
+  type Transaction   = Has[RocksDB.TransactionService]
+  type TransactionDB = Has[RocksDB.TransactionDBService]
+
+  object RocksDB {
+    trait TransactionService {
+      def get(readOptions: jrocks.ReadOptions, key: Bytes): Task[Option[Bytes]]
+      def getForUpdate(readOptions: jrocks.ReadOptions, key: Bytes, exclusive: Boolean): Task[Option[Bytes]]
+      def put(key: Bytes, value: Bytes): Task[Unit]
+      def delete(key: Bytes): Task[Unit]
+      def commit: Task[Unit]
+    }
+
+    trait TransactionDBService {
+      def beginTransaction(writeOptions: jrocks.WriteOptions): UIO[TransactionService]
+    }
+
+    def transaction[R <: Has[_], E >: Throwable, A](
+      zio: ZIO[Transaction with R, E, A]
+    ): ZIO[TransactionDB with R, E, A] = zio.provideSomeLayer[TransactionDB with R](ZTransaction.live)
+  }
+
+  // Helper Utilities
+  def get(readOptions: jrocks.ReadOptions, key: Bytes): RIO[Transaction, Option[Bytes]] =
```

Comment:
What’s the scope of the API of `Transaction` and `TransactionDB`? Are there any potential API conflicts? If so, we should move their accessors to different namespaces.

Reply:
`TransactionDB` extends `RocksDB` whereas `Transaction` doesn't have much in common with `RocksDB` or `TransactionDB`, merely a few `get` `put` APIs.

What do you recommend?